lib/fshash/metadata.go

```diff
@@ -0,0 +1,118 @@
+package fshash
+
+import (
+	"archive/tar"
+	"io"
+	"os"
+	"path/filepath"
+	"sort"
+	"time"
+
+	"github.com/spacemonkeygo/errors"
+	"github.com/ugorji/go/codec"
+	"polydawn.net/repeatr/def"
+)
+
+type Metadata tar.Header
+
+func ReadMetadata(path string, optional ...os.FileInfo) Metadata {
+	var fi os.FileInfo
+	var err error
+	if len(optional) > 0 {
```

Comment:
As of the tip of this branch, `ReadMetadata` is never called with `optional` params.
Yet it takes variadic args to `optional`, and further, only ever cares about the first one passed.

Unless there's something I'm missing, this should be changed to not be variadic.